dulwich/porcelain.py

```diff
@@ -3358,6 +3358,32 @@ def no_merged_branches(repo: RepoPath) -> Iterator[bytes]:
             yield branch_name
 
 
+def branches_containing(repo: RepoPath, commit: str) -> Iterator[bytes]:
+    """List branches that contain the specified commit.
+
+    Args:
+        repo: Path to the repository
+        commit: Commit-ish string (SHA, branch name, tag, etc.)
+
+    Yields:
+        Branch names (without refs/heads/ prefix) that contain the commit
+
+    Raises:
+        ValueError: If the commit reference is malformed
+        KeyError: If the commit reference does not exist
+    """
+    with open_repo_closing(repo) as r:
+        try:
+            commit_obj = parse_commit(r, commit)
+            commit_sha = commit_obj.id
+        except ValueError:
```

Comment:
Please drop this - it's not doing anything. You're catching the error and then just reraising it.